Query with REST API

brunopgalvao opened the pull request:
## 📝 Description

Resolves https://github.com/polkadot-developers/polkadot-docs/issues/1205

## 🔍 Review Preference

Choose one:
- [ ] ✅ I have time to handle formatting/style feedback myself 
- [x] ⚡ Docs team handles formatting (check "Allow edits from maintainers")  

## 🤖 AI-Ready Docs

If content changed, [regenerate AI files](../CONTRIBUTING.md#making-changes):
- [ ] ✅ I ran `python3 scripts/generate_llms.py`  
- [x] ⚡ Docs team will regenerate (check "Allow edits from maintainers")  

## ✅ Checklist

- [x] Changes tested  
- [ ] [PaperMoon Style Guide](https://github.com/papermoonio/documentation-style-guide) followed

nhussein11 commented on .chain-interactions/query-data/query-rest.md:
```diff
@@ -1 +1,255 @@
+| `sufficients` | Number of accounts whose existence is paid for by this asset |
+| `status` | Asset status (Live, Frozen, or Destroying) |
+
+## Query Block Information
```
I'd move this after foreign assets so it's more consistent; we show all assets-related actions first, then block-related actions.

brunopgalvao replied:
Done.

nhussein11 commented on .chain-interactions/query-data/query-rest.md:
```diff
@@ -1 +1,255 @@
+
+For production applications or high-frequency queries, run your own Sidecar instance.
+
+### Using npm
```
What about using tabs instead of ###?

nhussein11 commented on .chain-interactions/query-data/query-rest.md:
```diff
@@ -1 +1,255 @@
+
+For a complete list of endpoints and parameters, see the [Sidecar API Documentation](https://paritytech.github.io/substrate-api-sidecar/docsv2/){target=\_blank}.
+
+Common endpoints include:
```
I think this is redundant, considering all of them where explained above

brunopgalvao commented on .snippets/code/dummy/src/lib.rs:
```diff
@@ -0,0 +1,2 @@
+// Placeholder crate for CI workspace requirements.
```
Yes agreed, I plan on fixing this once we migrate the tutorial/guide code to the polkadot-cookbook, then we can get rid of this CI completely.

brunopgalvao replied:
As discussed, I removed the Cargo.toml and checks.yml in favor of the new test suite we are building in the polkadot-cookbook.

nhussein11 commented on .chain-interactions/query-data/query-rest.md:
````diff
@@ -1 +1,242 @@
+To query a specific asset (e.g., USDT with asset ID 1984):
+
+```bash
+curl -s "https://polkadot-asset-hub-public-sidecar.parity-chains.parity.io/accounts/<INSERT_ADDRESS>/asset-balances?assets[]=1984"
````
This appears to be a mix of placeholders and real values. I would use only placeholders and include the USDT bash call as an example in a collapsible element. @brunopgalvao, what do you think?

brunopgalvao replied:
This is good feedback. Updated to reflect this.

brunopgalvao commented:
> Hey @brunopgalvao, in the section Query Asset Details in the response fields, there are a few fields left to explain:
> 
> "pallet": "palletIndex": "storageItem": "keys"
> 
> Are they necessary to explain? Think this if you read only the section, not the entire doc. Also, in the Query Block Information and Query Asset Metadata sections, there is no response field. Is it necessary to add it here, too?

Hey @0xlukem good observation. I have updated the "Response Fields" to include a description of these keys. Given that, I would say that the fields which are of most importance for the developer for "Query Asset Details" and "Query Asset Metadata" are the ones returned in the `value` object.

Yes I agree with you @eshaben - each query should be encapsulated. It should currently be so.

For "Query Block Information", I do see a response from my end when I render the UI.

<img width="891" height="757" alt="Screenshot 2025-12-19 at 1 03 42 PM" src="https://github.com/user-attachments/assets/b7d5d284-4b02-4355-b7e5-84cabad1fe8e" />


Same for Query Asset Metadata.
<img width="906" height="674" alt="Screenshot 2025-12-19 at 1 04 21 PM" src="https://github.com/user-attachments/assets/66f2e8c9-c1f7-4623-8b0c-b6d3d6b59e4c" />


However I do believe these should be using termynal for the output. So I updated accordingly - should be good now.
